src/install/install.zig

```diff
@@ -1062,9 +1062,28 @@ pub const PackageInstall = struct {
         _ = package_json_checker.parseExpr() catch return false;
         if (!package_json_checker.has_found_name or !package_json_checker.has_found_version or log.errors > 0) return false;
 
-        // Version is more likely to not match than name, so we check it first.
-        return strings.eql(package_json_checker.found_version, this.package_version) and
-            strings.eql(package_json_checker.found_name, this.package_name);
+        // Check if the version matches
+        if (!strings.eql(package_json_checker.found_version, this.package_version)) {
+            // If it doesn't match, check again but with a leading "v" removed
+            //    v1.0.7
+            //     1.0.7
+            //    ^
+            //    ignore the v
```

Comment:
do we also need to check for `=`?
https://github.com/npm/node-semver/blob/816c7b2cbfcb1986958a290f941eddfd0441139e/internal/re.js#L124